### PR TITLE
cephfs: add functions for path and file Truncate

### PR DESCRIPTION
**Pull request by phlogistonjohn**
The cephfs packages gains new functions:
* File method Truncate implementing ceph_ftruncate
* Truncate implementing ceph_truncate

As a bonus, while adding the above I noted the coverage on path.go was nearly 100% but just missing an error condition check for Statx, so I added it here.

Fixes: #264 

## Checklist
- [x] Added tests for features and functional changes
- [x] Public functions and types are documented
- [x] Standard formatting is applied to Go code

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
Rebased. @ansiwen @nixpanic @agarwal-mudit PTAL

File: cephfs/file.go [hunk 1]
```diff
@@ -391,3 +391,25 @@ func (f *File) Fsync(sync SyncChoice) error {
...
+// Truncate sets the size of the open file.
+// NOTE: In some versions of ceph a bug exists where calling ftruncate on a
+// file open for read-only is permitted. The go-ceph wrapper does no additional
+// checking and will inherit the issue on affected versions of ceph.  Please
+// refer to the following issue for details:
+// https://tracker.ceph.com/issues/48202
+//
+// Implements:
+//  int ceph_ftruncate(struct ceph_mount_info *cmount, int fd, int64_t size);
+func (f *File) Truncate(size int64) error {
```
Comment (agarwal-mudit) [inline cephfs/file.go:404]: Any reason for not naming it Ftruncate, that's the nomenclature existing functions in the file have followed?

File: cephfs/path_test.go [hunk 2]
```diff
@@ -399,3 +399,52 @@ func TestRename(t *testing.T) {
 		assert.NoError(t, mount.RemoveDir(n2))
 	})
 }
+
+func TestTruncate(t *testing.T) {
```
Comment (agarwal-mudit) [inline cephfs/path_test.go:411]: Just curious, what will happen if we pass the path to a directory? Should we add a test case.

Comment (phlogistonjohn) [reply cephfs/path_test.go:411]: Ceph should error (with EBADF IIRC). I don't think it makes sense to cover every condition ceph libs _should_ check for. That said, if we find an issue we should file it like I did for the truncate-read-only-fd issue, which I stumbled on by accident.

File: cephfs/file.go (hunk 1 above)
Comment (phlogistonjohn) [reply cephfs/file.go:404]: When there's an established syscall interface with equivalent functionality (no alternative args, etc) that's part of Golang's file interface we've been using those names. This is such a case.

**Review by agarwal-mudit — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Pull request merged**